salt/modules/saltutil.py

```diff
@@ -78,15 +78,17 @@ def _get_top_file_envs():
     try:
         return __context__["saltutil._top_file_envs"]
     except KeyError:
+        st_ = salt.state.HighState(__opts__, initial_pillar=__pillar__.value())
```

Comment:
Is it worth turning `HighState` into a context manager?
How often do we use it?

Reply:
We use it enough that it does make sense to be a context manager. I've only changed this one place to actually use the context manager though.

Reply:
We'll address the other places in a future PR then.